rxandroid/src/main/java/rx/android/schedulers/AndroidSchedulers.java

```diff
@@ -38,4 +38,10 @@ public static Scheduler mainThread() {
                 RxAndroidPlugins.getInstance().getSchedulersHook().getMainThreadScheduler();
         return scheduler != null ? scheduler : MainThreadSchedulerHolder.MAIN_THREAD_SCHEDULER;
     }
+
+    public static Scheduler currentThread() {
```

Comment:
And at least two more tests: 

1. For a `Thread` with `Looper` to check that it picks its looper and runs the code on that thread.
2. For a `Thread` without `Looper` to see how it fails with `Can't create handler inside thread that has not called Looper.prepare()`.